In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df2 = pd.read_csv('/home/sneh.ratanpara@simform.dom/Sneh Ratanpara Learning/Python Programming/EDA/AQI_dataset/station_day.csv')

In [3]:
df2.sample(5)

,StationId,Date,PM2.5,PM10,NO,NO2,NOx,NH3,CO,SO2,O3,Benzene,Toluene,Xylene,AQI,AQI_Bucket
62381,KA005,2019-11-12,NaN,96.67,13.98,69.03,82.97,NaN,1.86,9.57,NaN,NaN,NaN,NaN,115.0,Moderate
68984,KA011,2019-09-28,21.12,52.96,5.34,16.00,13.68,9.46,0.80,1.51,29.77,0.07,0.57,NaN,84.0,Satisfactory
35070,DL024,2019-12-21,327.49,437.88,56.57,55.72,75.76,95.78,2.32,9.02,9.84,13.44,32.56,NaN,488.0,Severe
107331,WB013,2018-07-29,44.14,66.64,25.27,32.24,57.51,3.39,0.84,1.96,14.33,2.47,12.04,NaN,79.0,Satisfactory
41788,DL031,2015-05-25,112.32,399.47,83.58,78.82,157.42,17.64,2.10,11.98,21.34,1.17,3.54,NaN,315.0,Very Poor


In [8]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 108035 entries, 0 to 108034
Data columns (total 16 columns):
 #   Column      Non-Null Count   Dtype  
---  ------      --------------   -----  
 0   StationId   108035 non-null  object 
 1   Date        108035 non-null  object 
 2   PM2.5       86410 non-null   float64
 3   PM10        65329 non-null   float64
 4   NO          90929 non-null   float64
 5   NO2         91488 non-null   float64
 6   NOx         92535 non-null   float64
 7   NH3         59930 non-null   float64
 8   CO          95037 non-null   float64
 9   SO2         82831 non-null   float64
 10  O3          82467 non-null   float64
 11  Benzene     76580 non-null   float64
 12  Toluene     69333 non-null   float64
 13  Xylene      22898 non-null   float64
 14  AQI         87025 non-null   float64
 15  AQI_Bucket  87025 non-null   object 
dtypes: float64(13), object(3)
memory usage: 13.2+ MB


In [4]:
df2.describe()

,PM2.5,PM10,NO,NO2,NOx,NH3,CO,SO2,O3,Benzene,Toluene,Xylene,AQI
count,86410.000000,65329.000000,90929.000000,91488.000000,92535.000000,59930.000000,95037.000000,82831.000000,82467.000000,76580.000000,69333.000000,22898.000000,87025.000000
mean,80.272571,157.968427,23.123424,35.240760,41.195055,28.732875,1.605749,12.257634,38.134836,3.358029,15.345394,2.423446,179.749290
std,76.526403,123.418672,34.491019,29.510827,45.145976,24.897797,4.369578,12.984723,39.128004,11.156234,29.348587,6.472409,131.324339
min,0.020000,0.010000,0.010000,0.010000,0.000000,0.010000,0.000000,0.010000,0.010000,0.000000,0.000000,0.000000,8.000000
25%,31.880000,70.150000,4.840000,15.090000,13.970000,11.900000,0.530000,5.040000,18.895000,0.160000,0.690000,0.000000,86.000000
50%,55.950000,122.090000,10.290000,27.210000,26.660000,23.590000,0.910000,8.950000,30.840000,1.210000,4.330000,0.400000,132.000000
75%,99.920000,208.670000,24.980000,46.930000,50.500000,38.137500,1.450000,14.920000,47.140000,3.610000,17.510000,2.110000,254.000000
max,1000.000000,1000.000000,470.000000,448.050000,467.630000,418.900000,175.810000,195.650000,963.000000,455.030000,454.850000,170.370000,2049.000000


In [5]:
df2.isnull().sum()

StationId         0
Date              0
PM2.5         21625
PM10          42706
NO            17106
NO2           16547
NOx           15500
NH3           48105
CO            12998
SO2           25204
O3            25568
Benzene       31455
Toluene       38702
Xylene        85137
AQI           21010
AQI_Bucket    21010
dtype: int64

In [10]:
df2.isnull().mean()*100

StationId      0.000000
Date           0.000000
PM2.5         20.016661
PM10          39.529782
NO            15.833758
NO2           15.316333
NOx           14.347202
NH3           44.527237
CO            12.031286
SO2           23.329477
O3            23.666404
Benzene       29.115564
Toluene       35.823576
Xylene        78.805017
AQI           19.447401
AQI_Bucket    19.447401
dtype: float64

## Conclusions

**Xylene column have 78.8% missing values so it is better to drop that column**

**This table is more larger tha nthe city_day table**

**NH# and PM10 columns have high missing values than city_day table**

**Need to merge this on station table to get more useful**